test/jdk/java/lang/invoke/common/test/java/lang/invoke/lib/InstructionHelper.java

```diff
@@ -23,121 +23,104 @@
+    private static void commonBuild(ClassBuilder classBuilder) {
+        classBuilder
+                .withVersion(55, 0)
+                .withSuperclass(ConstantDescs.CD_Object)
+                .withMethod(ConstantDescs.INIT_NAME, ConstantDescs.MTD_void, Classfile.ACC_PUBLIC,
+                        methodBuilder -> methodBuilder
+                                .withCode(codeBuilder -> codeBuilder
+                                        .aload(0)
+                                        .invokespecial(ConstantDescs.CD_Object, ConstantDescs.INIT_NAME,
+                                                ConstantDescs.MTD_void, false)
+                                        .return_()));
     }
 
-    public static MethodHandle ldcDynamicConstant(MethodHandles.Lookup l,
-                                                  String name, Class<?> type,
-                                                  String bsmMethodName, MethodType bsmType,
-                                                  Consumer<PoolHelper.StaticArgListBuilder<String, String, byte[]>> staticArgs) throws Exception {
-        return ldcDynamicConstant(l, name, type, l.lookupClass(), bsmMethodName, bsmType, staticArgs);
+    private static String className(MethodHandles.Lookup l) {
+        return l.lookupClass().getCanonicalName().replace('.', '/') + "$Code_" +
+                COUNT.getAndIncrement();
     }
 
-    public static MethodHandle ldcDynamicConstant(MethodHandles.Lookup l,
-                                                  String name, Class<?> type,
-                                                  Class<?> bsmClass, String bsmMethodName, MethodType bsmType,
-                                                  Consumer<PoolHelper.StaticArgListBuilder<String, String, byte[]>> staticArgs) throws Exception {
-        return ldcDynamicConstant(l, name, cref(type), csym(bsmClass), bsmMethodName, bsmType.toMethodDescriptorString(), staticArgs);
+    public static MethodHandle invokedynamic(MethodHandles.Lookup l, String name, MethodType type, String bsmMethodName,
+                                             MethodType bsmType,
+                                             ConstantDesc[] boostrapArgs) throws Exception {
+        byte[] byteArray = Classfile.of().build(ClassDesc.of(className(l)), classBuilder -> {
+            commonBuild(classBuilder);
+            classBuilder
+                    .withMethod("m", MethodTypeDesc.ofDescriptor(type.toMethodDescriptorString()),
+                            Classfile.ACC_PUBLIC + Classfile.ACC_STATIC, methodBuilder -> methodBuilder
+                                    .withCode(codeBuilder -> {
+                                        for (int i = 0; i < type.parameterCount(); i++) {
+                                            codeBuilder.loadInstruction(TypeKind.fromDescriptor(
+                                                    type.parameterType(i).descriptorString()), i);
```

Comment:
You can just use `TypeKind.from(type.parameterType(i))`

Reply:
I can't see such as method.

Reply:
It is declared as `public static TypeKind from(TypeDescriptor.OfField<?> descriptor)` so that you can pass either `Class` or `ClassDesc` in.